plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -271,8 +253,301 @@ type Query {
 }
 `;
 
-// Pass implementation of MetronomeClient to the resolvers for easier testability
-export const resolvers = (): IResolvers => ({
+interface ResolverArgs {
+  fetchJobDetail: (id: string) => Observable<MetronomeClient.JobDetailResponse>;
+  pollingInterval: number;
+}
+
+function isActiveRun(arg: any): arg is MetronomeClient.ActiveJobRun {
+  return arg.jobId !== undefined;
+}
+
+function isHistoryRun(arg: any): arg is MetronomeClient.JobHistoryRun {
+  return arg.id !== undefined;
+}
+
+const typeResolvers = {
```

Comment:
Moving forward I kind think we want to separate those type resolvers per their own model files. So we don't have a high churning 1k lines code. 

I think the graphql part of the data layer needs a lot of improvements in making our life easier.

Reply:
Yeah, I would like to do a second PR somewhen soon to move the type resolvers into the normal resolvers file. For the file size, I think we should separate at least the graphql and the TS definitions derived from that, this would help a lot already. If it's still bad, maybe let's split it up by type?